lib/srv6.c

```diff
@@ -114,3 +125,53 @@ const char *seg6local_context2str(char *str, size_t size,
 		return str;
 	}
 }
+
+struct srv6_locator *srv6_locator_alloc(const char *name)
+{
+	int namelen = 0;
+	struct srv6_locator *locator = NULL;
+
+	locator = XCALLOC(MTYPE_SRV6_LOCATOR, sizeof(struct srv6_locator));
+	if (locator) {
+		namelen = sizeof(locator->name);
+		if (namelen > SRV6_LOCNAME_SIZE - 1)
+			namelen = SRV6_LOCNAME_SIZE - 1;
+		strlcpy(locator->name, name, namelen);
+		locator->current = 0;
```

Comment:
This doesn't need to be set to 0, you allocated the struct with a clearing allocator (XCALLOC).

Reply:
I fixed it